### ISSUE TITLE
Weird definition of overlap

### ISSUE THREAD

**Issue by Darksonn** [rust-lang/fls#372]
You define overlap as

> Two [values](https://spec.ferrocene.dev/glossary.html#term_value) overlap when their memory locations overlap, or both values are elements of the same [array](https://spec.ferrocene.dev/glossary.html#term_array).

So even if they don't overlap in memory location, but are part of the same array, they are still considered to overlap. However, you then define:

> While a [mutable reference](https://spec.ferrocene.dev/glossary.html#term_mutable_reference) is [active](https://spec.ferrocene.dev/ownership-and-deconstruction.html?highlight=overlap#term_active), no other [reference](https://spec.ferrocene.dev/glossary.html#term_reference) shall refer to a [value](https://spec.ferrocene.dev/glossary.html#term_value) that [overlaps](https://spec.ferrocene.dev/glossary.html#term_overlap) with the [referent](https://spec.ferrocene.dev/glossary.html#term_referent) of the [mutable reference](https://spec.ferrocene.dev/glossary.html#term_mutable_reference).

But mutable references to arrays can usually be split into several references if they don't overlap in the conventional sense, but this contradicts the definition with arrays.

**Comment by jonas-schievink** [comment]
Yeah, these rules only apply when borrow checking manually created references, so we should clarify that. `&mut a[0]` and `&mut a[1]` conflict, but `let [x,y,..] = a` is fine (and so is an unsafe implementation like [`split_at_mut`](https://doc.rust-lang.org/std/primitive.slice.html#method.split_at_mut)).

**Comment by PLeVasseur** [comment]
Hi @Darksonn 👋 do you still feel this is imprecise enough to warrant some attention?
